docs/making-a-module.md

```diff
@@ -0,0 +1,179 @@
+# Making a Module
+Before making a module, follow the 
+[Getting Started](./getting-started.md) guide to setup development.
+
+## Table of contents
+* [Module setup](#module-setup)
+  * [Beet config](#beet-config)
+  * [Load](#load)
+* [Writing the module](#writing-the-module)
+  * [Guidebook pages](#guidebook-pages)
+  * [Tests](#tests)
+* [Submitting a pull request](#submitting-a-pull-request)
+* [Finishing up for release](#finishing-up-for-release)
+  * [Credits](#credits)
+  * [Readme](#readme)
+  * [Icon](#icon)
+  * [Images](#images)
+  * [Readme](#readme)
+  * [Wiki page](#wiki-page)
+
+## Module setup
+As with any changes to the repository, create a new branch:
+```
+git checkout -b my-new-module
+```
+
+Start by copying the `gm4_bat_grenades` folder to the new module ID. It is by far the smallest module so it's easy to use as a template.
+* Rename the folder `gm4_bat_grenades/data/gm4_bat_grenades` to `gm4_<module_id>/data/gm4_<module_id>`.
+* In the `functions` folder, find and replace all instances of `bat_grenades` with `<module_id>`.
+
+### Beet config
+Each module has a `beet.yaml` file that contains information on how to build the pack and some metadata used on the website.
+```yaml
+# The ID and name of this module. The ID should always be the same as the current directory name
+id: gm4_disassemblers
+name: Disassemblers
+# Version "patch" value is managed by a github action. Leave it as 'X'
+version: '1.3.X'
+
+# Load the current directory
+data_pack:
+  load: .
+
```

Comment:
Should also load the resource pack if there are any files (unless RP isnt a part of this PR intentionally)

```
resource_pack:
    load: .
```

Reply:
Since the #918 is not merged yet I didn't include any of it yet. There will need to be a separate `docs/resource-pack.md` file and some changes in the other docs.